src/hooks/useSpruceConfig.ts

```diff
@@ -5,11 +5,12 @@ import {
 } from "gql/generated/types";
 import { SPRUCE_CONFIG } from "gql/queries";
 
-export const useSpruceConfig = () => {
+export const useSpruceConfig = ():
+  | SpruceConfigQuery["spruceConfig"]
+  | undefined => {
   const { data } = useQuery<SpruceConfigQuery, SpruceConfigQueryVariables>(
     SPRUCE_CONFIG
   );
 
-  const { spruceConfig } = data || {};
-  return spruceConfig;
+  return data?.spruceConfig;
```

Comment:
I am kinda confused about how this will fix the flakiness. I thought the optional chaining in e.g. `GeneralTab` would handle receiving an empty object, do you know why that hasn't been working as expected?